src/azure-cli/azure/cli/command_modules/appconfig/azconfig/models.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from enum import Enum
+import uuid
+import azure.cli.command_modules.appconfig.azconfig.constants as constants
+
+
+# pylint: disable=too-few-public-methods
+# pylint: disable=too-many-instance-attributes
+
+class KeyValue(object):
+    '''
+    Key value class.
+
+    :ivar str key:
+        Key of the entry.
+    :ivar str value:
+        Value of the entry.
+    :ivar str label:
+        Label of the entry.
+    :ivar dict[str, str] tags:
+        Dictionary of tags of the entry.
+    :ivar str content_type:
+        Content type of the entry.
+    :ivar str etag:
+        The ETag contains a value that you can use to perform operations.
+    :ivar bool locked:
+        Represents whether the key value entry is locked.
+    :ivar datetime last_modified:
+        A datetime object representing the last time the key was modified.
+    '''
+
+    def __init__(self, key, value=None, label=None, tags=None, content_type=None):
+        self.key = key
+        self.value = value
+        self.label = label
+        self.tags = tags
+        self.content_type = content_type
+        self.etag = None
+        self.last_modified = None
+        self.locked = None
+
+
+class QueryFields(Enum):
+    KEY = 0x001
+    LABEL = 0x002
+    VALUE = 0x004
+    CONTENT_TYPE = 0x008
+    ETAG = 0x010
+    LAST_MODIFIED = 0x020
+    LOCKED = 0x040
+    TAGS = 0x080
+    ALL = KEY | LABEL | VALUE | CONTENT_TYPE | ETAG | LAST_MODIFIED | LOCKED | TAGS
+
+
+class QueryKeyValueOptions(object):
+    '''
+    Query options for retriving the key-value
+
+    :ivar str label:
+        Label of the key-value entry.
+    :ivar datetime query_datetime:
+        A datetime object representing a timestamp when the request is going to hit.
+    :ivar list of QueryFields:
+        Client side filter of the query.
+    :ivar str client_request_id:
+        A request ID that, if provided, can be used to help track the operation.
+    :ivar string correlation_request_id:
+        An ID that can be used to correlate the request with a more general operation.
+    '''
+
+    empty_label = u'\0'
+
+    def __init__(self,
+                 label=empty_label,
+                 query_datetime=None,
+                 fields=None,
+                 client_request_id=None,
+                 correlation_request_id=None):
+        self.label = label
+        self.query_datetime = query_datetime
+        self.fields = fields
+        self.client_request_id = str(
+            uuid.uuid4()) if client_request_id is None else client_request_id
+        self.correlation_request_id = str(
+            uuid.uuid4()) if correlation_request_id is None else correlation_request_id
+
+
+class QueryKeyValueCollectionOptions(object):
+    '''
```

Comment:
Is this your SDK?  It should not be embedded in the CLI...

Reply:
It is our internally used SDK. We don't plan to publish it and instead we will leverage Azure SDK team to ship our data plane Python SDK. And they are still working on it.